src/static/js/routes/index.js

```diff
@@ -7,193 +7,245 @@
-        y: radius
-      };
+  _radius = $chartsContainerDom.width() * _multiplier / 2;
+  _arcStart = 270;
+  _center = {
+    x: _radius,
+    y: _radius
+  };
 
   // Array of objects to store each chart's values.
-  var charts = [
-    {
-      name:  'consumer guides',
-      done:  parseInt( $( '#consumer-guides .chart-value' ).text(), 10 ) / 100,
-      value: 0
-    },
-    {
-      name:  'submit a complaint',
-      done:  parseInt( $( '#submit-a-complaint .chart-value' ).text(), 10 ) / 100,
-      value: 0
-    },
-    {
-      name:  'data and research',
-      done:  parseInt( $( '#data-and-research .chart-value' ).text(), 10 ) / 100,
-      value: 0
-    },
-    {
-      name:  'policy and compliance',
-      done:  parseInt( $( '#policy-and-compliance .chart-value' ).text(), 10 ) / 100,
-      value: 0
-    },
-    {
-      name:  'about',
-      done:  parseInt( $( '#about .chart-value' ).text(), 10 ) / 100,
-      value: 0
-    }
-  ];
+  _charts = _buildCharts();
+
+  _$chartsDom = $( '.progress-charts_chart' );
 
   // Set up each canvas element.
-  $( '.progress-charts_chart' ).each( function() {
+  _$chartsDom.each( function() {
     var ctx = $( this )[0].getContext( '2d' );
-    ctx.canvas.width = $( '.progress-charts_container' ).width() * multiplier;
+    ctx.canvas.width = $chartsContainerDom.width() * _multiplier;
     ctx.canvas.height = ctx.canvas.width;
   } );
 
-  // Run everything on load (or switch to when in view).
-  $( document ).ready( function() {
-    // TODO: Once page-specific JS is only run on the homepage
-    // this check will not be necessary any longer.
-    var chartAvail = $( '.progress-charts_chart' );
-    if ( chartAvail && !animationComplete ) {
-      animate();
-    }
-  } );
+  _animate();
+}
 
-  function animate() {
-    // Variable to keep track of when all charts are finished animating.
-    var completeCount = 0;
-    $( '.progress-charts_chart' ).each( function( i ) {
-      // snag the context
-      var ctx = $( this )[0].getContext( '2d' );
-      if ( charts[i].value < charts[i].done ) {
-        // ease the value (90 being how long it should take)
-        // we're animating at 60fps so 90 is 1.5 seconds
-        charts[i].value = easeInOutQuint( iterationCount, 0, charts[i].done, 90 );
-      } else {
-        completeCount++;
-      }
-      // send values to be drawn
-      drawGraph( ctx, charts[i].value );
+/**
+ * Build the source charts data based on the value set in the text in the DOM.
+ * @returns {Array} List of objects with done and value settings corresponding
+ *   to each graph item, where done is the total amount to draw,
+ *   and value is where to start animating from.
+ */
+function _buildCharts() {
+  var chartsValueSelector = '.progress-charts_container .chart-value';
+  var chartsDom = document.querySelectorAll( chartsValueSelector );
+  var charts = [];
+  var chartItem;
+  var doneValue;
+
+  for ( var c = 0, len = chartsDom.length; c < len; c++ ) {
```

Comment:
I don't believe this is checked every time. It would be if it was `for ( var c = 0; c < chartsDom.length; c++ ) {`. I picked this syntax up from High Performance JavaScript by @nzakas